Velocity_Model/z.py

```diff
@@ -0,0 +1,173 @@
+    :param nzvm_path:
+    :param version: NZVM version to extract values for
+    :return: Dataframe containing all z_types, station names and sigma
+    """
+    with tempfile.TemporaryDirectory() as temp_dir:
+        config_ffp = os.path.join(temp_dir, "GENERATE_THRESHOLD_USER_INPUT.txt")
+        coords_ffp = os.path.join(temp_dir, "MultipleProfileParameters.txt")
+        vm_working_dir = os.path.join(temp_dir, "z_output")
+
+        with open(coords_ffp, "w") as coords_fp:
+            coords_fp.write(f"{len(stat_df)}\n")
+        stat_df["stat_name"] = stat_df.index
+        stat_df[["lon", "lat", "stat_name"]].to_csv(
+            coords_ffp, sep=" ", header=None, mode="a", index=None
+        )
+
+        merged_df = stat_df
+        for z_type in z_types:
+            z_values = calculate_z(
+                config_ffp, coords_ffp, nzvm_path, version, vm_working_dir, z_type
+            )
+            stat_df["z_index"] = z_values.index
+            merged_df = merged_df.merge(z_values, left_on="z_index", right_index=True)
+
+    merged_df["sigma"] = calculate_z_sigma(stat_df, version)
+
+    output_columns = [column_names[z_type] for z_type in z_types] + ["sigma"]
+    merged_df.index.name = "Station_Name"
+    return merged_df[output_columns]
+
+
+def calculate_z(config_ffp, coords_ffp, nzvm_path, version, vm_working_dir, z_type):
+    """
+    For a given Z type and coords file write a config file and calculate Z by calling the NZVM binary
+
+    :param config_ffp: Path to config file to be written
+    :param coords_ffp: Path to given coords file
+    :param nzvm_path: path to the NZVM binary
+    :param version: version of the nzvm binary to extract the values for
+    :param vm_working_dir: folder where the raw output is written
+    :param z_type: Z1.0 or Z2.5
+    :return: Dataframe containing Z values and lat/lon
+    """
+
+    with open(config_ffp, "w") as config_fp:
+        config_fp.write(
+            "CALL_TYPE=GENERATE_THRESHOLD_USER_INPUT\n"
+            f"MODEL_VERSION={version}\n"
+            f"OUTPUT_DIR={vm_working_dir}\n"
+            f"COORDINATES_TEXTFILE={coords_ffp}\n"
+            f"VS_TYPE={z_type}\n"
+        )
+    shutil.rmtree(vm_working_dir, ignore_errors=True)
+    subprocess.run([nzvm_path, config_ffp], cwd=nzvm_path.parent)
+    z_values = pd.read_csv(
+        os.path.join(vm_working_dir, file_output_format[z_type]),
+        delim_whitespace=True,
+    )
+
+    return z_values
+
+
+def calculate_z_sigma(stat_df, version):
+    """
+    Returns a np array of len(stat_df) containing 0.3 or 0.5 depending if the station is within a basin or not
+
+    Loops over the respective basin files for each NZVM version
+    """
+    sigma_df = stat_df.copy()
+    sigma_df["in_basin_flag"] = False
+    sigma_df = sigma_df[
+        ["in_basin_flag"]
+    ]  # drops all columns other than the flag and the index
```

Comment:
sigma_df = pd.DataFrame(index=stat_df.index)
sigma_df["in_basin_flag"] = False

Would be more efficient than copying the whole dataframe then dropping most of it? Or at least less lines?